src/browser/modules/Stream/CypherFrame/index.jsx

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { connect } from 'preact-redux'
+import { Component } from 'preact'
+import FrameTemplate from '../FrameTemplate'
+import { CypherFrameButton } from 'browser-components/buttons'
+import Centered from 'browser-components/Centered'
+import bolt from 'services/bolt/bolt'
+import { deepEquals } from 'services/utils'
+import FrameSidebar from '../FrameSidebar'
+import { VisualizationIcon, TableIcon, AsciiIcon, CodeIcon, PlanIcon, AlertIcon, ErrorIcon, Spinner } from 'browser-components/icons/Icons'
+import { AsciiView, AsciiStatusbar } from './AsciiView'
+import { TableView, TableStatusbar } from './TableView'
+import { CodeView, CodeStatusbar } from './CodeView'
+import { ErrorsViewBus as ErrorsView, ErrorsStatusbar } from './ErrorsView'
+import { WarningsView, WarningsStatusbar } from './WarningsView'
+import { PlanView, PlanStatusbar } from './PlanView'
+import { VisualizationConnectedBus } from './VisualizationView'
+import Render from 'browser-components/Render'
+import Display from 'browser-components/Display'
+import * as viewTypes from 'shared/modules/stream/frameViewTypes'
+import { resultHasRows, resultHasWarnings, resultHasPlan, resultIsError, resultHasNodes, initialView } from './helpers'
+import { StyledFrameBody, SpinnerContainer, StyledStatsBarContainer } from '../styled'
+import { getMaxRows, getInitialNodeDisplay, getMaxNeighbours, shouldAutoComplete } from 'shared/modules/settings/settingsDuck'
+
+export class CypherFrame extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      openView: undefined,
+      exportData: null
+    }
+  }
+  changeView (view) {
+    this.setState({openView: view})
+    if (this.props.onRecentViewChanged) {
```

Comment:
This onRecentViewChanged hook was useful in CypherFrame because it did not had access to redux or bus, but since now this is a connected component I think we can directly dispatch this action from this component.